### PR TITLE
CTX-4993: Model comparison template implementation.

### PR DESCRIPTION
**Pull request by bogdant36**
CTX-4993: Model comparison template implementation.

### CODE REVIEW

File: tasks/model-comparison/main.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+    models: list[Model] = taskRun.parameters["models"]
+    logging.info(">> [Model Comparison] Fetching models and starting comparison.")
+
+    if models is None:
+        raise ValueError(">> [Model Comparison] Empty list of models provided.")
+
+    logging.info(">> [Model Comparison] Searching for most accurate model...")
+    bestModel = max(models, key = lambda x: x.accuracy)
+    logging.info(f">> [Model Comparison] Most accurate model found: {bestModel.name}")
```
Comment (dule1322) [inline tasks/model-comparison/main.py:17]: Instead of `:` use `"` to wrap model name

File: tasks/model-comparison/main.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+    logging.info(">> [Model Comparison] Starting model submit.")
```
Comment (dule1322) [inline tasks/model-comparison/main.py:19]: Submitting model as output...

File: tasks/model-comparison/main.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    taskRun.submitOutput("outputModel", bestModel)
+    logging.info(">> [Model Comparison] Model submitted successfully!")
```
Comment (dule1322) [inline tasks/model-comparison/main.py:21]: Model submitted

File: tasks/model-comparison/task.yaml [hunk 4]
```diff
@@ -0,0 +1,20 @@
+id: adf92e59-35a6-47b9-b526-0d6615d74b4e
+name: Model Comparison
+description: Compares provided Coretex models
```
Comment (dule1322) [inline tasks/model-comparison/task.yaml:3]: Add more detailed explanation of what this task does

File: tasks/model-comparison/main.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+from coretex import currentTaskRun, Model
+
+
+def main() -> None:
+    taskRun = currentTaskRun()
+
+    models: list[Model] = taskRun.parameters["models"]
+    logging.info(">> [Model Comparison] Fetching models and starting comparison.")
+
+    if models is None:
```
Comment (dule1322) [inline tasks/model-comparison/main.py:12]: There is no need for this check, models type is marked as `list[Model]` and the parameter is required to exist in task.yaml

File: tasks/model-comparison/main.py [hunk 6]
```diff
@@ -0,0 +1,25 @@
+import logging
+
+from coretex import currentTaskRun, Model
+
+
+def main() -> None:
+    taskRun = currentTaskRun()
+
+    models: list[Model] = taskRun.parameters["models"]
+    logging.info(">> [Model Comparison] Fetching models and starting comparison.")
```
Comment (dule1322) [inline tasks/model-comparison/main.py:10]: this log can be deleted, list of models is already fetched even before the main method is executed

**Review by dule1322 — CHANGES_REQUESTED** [review]

File: tasks/model-comparison/task.yaml [hunk 7]
```diff
@@ -0,0 +1,22 @@
+id: adf92e59-35a6-47b9-b526-0d6615d74b4e
+name: Model Comparison
+description:
+  "Task that takes a list of Coretex models as input and compares them by their accuracy.\n
```
Comment (dule1322) [inline tasks/model-comparison/task.yaml:4]: First line of description should be on the same line as the field definition.
There is no need to manually enter a "\n" character. Yaml format supports them by default

Comment (bogdant36) [reply tasks/model-comparison/task.yaml:4]: Auto save in VSC automatically add's those lines, but will try to edit it in simple text editor instead.

**Review by dule1322 — APPROVED** [review]

**Pull request merged**